docs/typescript/schemas.md

```diff
@@ -1,89 +1,86 @@
+
+If automatic type inference doesn't work for you, you can always fall back to document interface definitions.
 
-Mongoose can also automatically infer the document type from your schema definition as follows.
+## Separate document interface definition
+
+If automatic type inference doesn't work for you, you can define a separate raw document interface as follows.
 
 ```typescript
-import { Schema, InferSchemaType } from 'mongoose';
+import { Schema } from 'mongoose';
 
-// Document interface
-// No need to define TS interface any more.
-// interface User {
-//   name: string;
-//   email: string;
-//   avatar?: string;
-// }
+// Raw document interface. Contains the data type as it will be stored
+// in MongoDB. So you can ObjectId, Buffer, and other custom primitive data types.
+// But no Mongoose document arrays or subdocuments.
+interface User {
+  name: string;
+  email: string;
+  avatar?: string;
+}
 
 // Schema
-const schema = new Schema({
+const schema = new Schema<User>({
   name: { type: String, required: true },
   email: { type: String, required: true },
   avatar: String
 });
-
-type User = InferSchemaType<typeof schema>;
-// InferSchemaType will determine the type as follows:
-// type User = {
-//   name: string;
-//   email: string;
-//   avatar?: string;
-// }
-
-// `UserModel` will have `name: string`, etc.
-const UserModel = mongoose.model('User', schema);
 ```
 
-There are a few caveats for using automatic type inference:
-
-1. You need to set `strictNullChecks: true` or `strict: true` in your `tsconfig.json`. Or, if you're setting flags at the command line, `--strictNullChecks` or `--strict`. There are [known issues](https://github.com/Automattic/mongoose/issues/12420) with automatic type inference with strict mode disabled.
-2. You need to define your schema in the `new Schema()` call. Don't assign your schema definition to a temporary variable. Doing something like `const schemaDefinition = { name: String }; const schema = new Schema(schemaDefinition);` will not work.
-3. Mongoose adds `createdAt` and `updatedAt` to your schema if you specify the `timestamps` option in your schema, *except* if you also specify `methods`, `virtuals`, or `statics`. There is a [known issue](https://github.com/Automattic/mongoose/issues/12807) with type inference with timestamps and methods/virtuals/statics options. If you use methods, virtuals, and statics, you're responsible for adding `createdAt` and `updatedAt` to your schema definition.
-
-If automatic type inference doesn't work for you, you can always fall back to document interface definitions.
+By default, Mongoose does **not** check if your raw document interface lines up with your schema.
+For example, the above code won't throw an error if `email` is optional in the document interface, but `required` in `schema`.
 
 ## Generic parameters
 
 The Mongoose `Schema` class in TypeScript has 4 [generic parameters](https://www.typescriptlang.org/docs/handbook/2/generics.html):
```

Comment:
```suggestion
The Mongoose `Schema` class in TypeScript has 9 [generic parameters](https://www.typescriptlang.org/docs/handbook/2/generics.html):
```

if counting all the generics in the current schema type, unless i missed something

https://github.com/Automattic/mongoose/blob/5c4051ac56b4414afe7a6a02f6b946c7360c5ea8/types/index.d.ts#L219-L235